### PR TITLE
Add CloudFormation template for creating AWS resources needed by the build-bundle-test workflows

### PR DESCRIPTION
**Pull request by camerski**
### Description
CloudFormation stack for managing the S3 bucket used by the build-bundle-test workflows and IAM users for each of the three stages.
 
### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/143
 
### Testing
Manually verified that the bucket and IAM users are created. Tested in the IAM policy simulator to verify that the users have the correct access to folders within the bucket.

### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: cloudformation/build-bundle-test-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+Description: Template for setting up resources for the build-bundle-test workflows
+Resources:
+  BuildBucket:
+    Type: "AWS::S3::Bucket"
+  BuildUser:
+    Type: "AWS::IAM::User"
```
Comment (peternied) [inline cloudformation/build-bundle-test-workflow.yml:6]: Why are we creating users?  Won't the scripts be run inside of the Jenkins system, or on a developer machine where they could assume a specific role.

File: cloudformation/build-bundle-test-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    Properties:
+      UserName: builder
+      Policies:
+        - PolicyName: S3_Write_Access
```
Comment (peternied) [inline cloudformation/build-bundle-test-workflow.yml:10]: Rename to `S3_Build_Write_Access` for consistency.

File: cloudformation/build-bundle-test-workflow.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+              Effect: Allow
+              Action: "s3:PutObject"
+              Resource:
+                - !Sub "arn:aws:s3:::${BuildBucket}/builds/*"
+  BundleUser:
+    Type: "AWS::IAM::User"
+    Properties:
+      UserName: bundler
+      Policies:
+        - PolicyName: S3_Build_Read_Access
```
Comment (peternied) [inline cloudformation/build-bundle-test-workflow.yml:23]: Aren't these already public buckets - is granting read access needed?

**Review by peternied — CHANGES_REQUESTED** [review]
Rather than have the CFN yml, what about making a fresh CDK project with multiple stacks to manage this first of many changes?

* Requesting changes on Users v Roles comment

File: cloudformation/build-bundle-test-workflow.yml (hunk 1 above)
Comment (mch2) [reply cloudformation/build-bundle-test-workflow.yml:6]: We would like to trigger the bundle builds & assembly fro Github Workflows which requires a User.

I don't think we need two sets of these creds though.  Perhaps we create a single user that is allowed to assume both these roles.

Comment (camerski) [reply cloudformation/build-bundle-test-workflow.yml:6]: Yeah, I'll switch to roles and create a single "system" user that can have credentials in the CI system.

File: cloudformation/build-bundle-test-workflow.yml (hunk 3 above)
Comment (camerski) [reply cloudformation/build-bundle-test-workflow.yml:23]: Only the prod bucket will be public. For our beta stack and individual developer stacks everything should be private. Also, IAM roles still need permissions to invoke GetObject (S3's permissions model is more complicated than other AWS services...access is gated by both the bucket policy and the caller's permissions).

File: cloudformation/build-bundle-test-workflow.yml (hunk 1 above)
Comment (camerski) [reply cloudformation/build-bundle-test-workflow.yml:6]: Done

File: cloudformation/build-bundle-test-workflow.yml (hunk 2 above)
Comment (camerski) [reply cloudformation/build-bundle-test-workflow.yml]: Done

**Review by peternied — APPROVED** [review]

**Review by mch2 — APPROVED** [review]

**Pull request merged**